pep/app/pods/components/modal-dialogs/user/preferences/template.hbs

```diff
@@ -33,6 +33,7 @@
                         @placeholder={{t "preferences.search.hic.limit"}}
                         @value={{this.currentUser.preferences.searchHICLimit}}
                         maxlength="2"
+                        {{on "input" (fn this.updatePreference this.hicLimit)}}
```

Comment:
since these updates are hitting the api if you are logged in we should at least debounce these updates from free text fields with a concurrency task I think. Otherwise switch to a changeset/Save button flow for saving updates rather than updating as input values change.

Reply:
Not sure if thats really necessary since the field has a `maxlength` of 2. I could see that being a nice to have down the line if we have extra time?

Reply:
thats true, its not a _huge_ deal here (though you'd still get excess requests if the user types in like "10", then backspaces to "1" -- 3 api requests instead of 1 if debounced), but i'd just hate to get in that habit or moreso have patterns in place that other devs who come onto the project might copy in places where it definitely is more necessary.

Plus I mean its very low hanging fruit, would just add a couple more lines to the component if we did something like:
```ts
@restartableTask
*debouncedUpdatePreference(/*...*/) {
    yield timeout(250);
    yield this.updatePreference(/*...*/);
}
```
```hbs
{{on "input" (perform this.debouncedUpdatePreference this.hicLimit)}}
```